vmm/src/interrupt.rs

```diff
@@ -241,22 +242,93 @@ impl InterruptSourceGroup for MsiInterruptGroup {
     }
 }
 
+pub struct LegacyInterruptGroup {
```

Comment:
We may consider calling that one `LegacyUserspaceInterruptGroup` because this one is tightly related to our userspace IOAPIC implementation.